baseband/vdif/base.py

```diff
@@ -124,7 +124,8 @@ def find_header(self, template_header=None, framesize=None, edv=None,
                 if template_header.same_stream(header):
                     self.seek(-header.size, 1)
                     return header
-            # Didn't work, so get searching.
+            # Didn't work, so reset the file pointer and get searching.
+            self.seek(-template_header.size, 1)
```

Comment:
This is not guaranteed to be correct -- we might have been close to the end of file, in which case fewer bytes were read. This is not really going to matter, but could you change it to just store the current offset before the `try`, and then reset it to that value? You could do that in a `finally` clause (which always gets executed, even if the `return` path was taken, so the seek there could be removed as well).